pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -201,7 +201,7 @@ func NewAuthCreds(refHostname string) (AuthCreds, error) {
 			// - ac.ServerAddress:         "https://index.docker.io/v1/".
 			if !isAuthConfigEmpty(ac) {
 				if authConfigHostname == registry.IndexServer {
```

Comment:
`ac.ServerAddress = registry.IndexServer`

I think we should enforce the default registry   when it is empty for all `refHostname`